File: dynamic programming/maxSubarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int numsSize = nums.size();
        int maxSubArrays[numsSize] = {0};
        int max = nums[0];

        for(int i=0 ; i < numsSize ;++i){

        	for(int j=0; j <= i ; ++j){

        		maxSubArrays[j] += nums[i];
        		if(maxSubArrays[j] > max)
        			max = maxSubArrays[j];
        	}
        }
        return max;
    }
};

File: Data Structures Part 1 Questions/maxSubArray.go
func maxSubArray(nums []int) int {
    intSize := 32
    max := -1 << (intSize - 1)
    cur := 0
    for _, i := range nums {
        if cur < 0 {
            cur = 0
        }
        cur += i
        if cur > max {
            max = cur
        }
    }
    return max
}